File: src/main/java/org/featx/jvmon/entity/AbstractIdentifiable.java
package org.featx.jvmon.entity;

import lombok.Data;
import java.util.Optional;

/**
 * @author Excepts
 */
@Data
public abstract class AbstractIdentifiable<I> implements Identifiable<I> {

    private I id;

    private Boolean deleted;

    @Override
    public Boolean isNew() {
        return !Optional.of(id).isPresent();
    }
}

File: src/main/kotlin/org/featx/jvmon/entity/AbstractIdentifiable.kt
package org.featx.jvmon.entity

import lombok.Data
import lombok.EqualsAndHashCode

import java.util.Optional

@Data
@EqualsAndHashCode(callSuper = true)
abstract class AbstractIdentifiable<I> : Identifiable<I> {
    override var id: I? = null
        set(id) {
            field = this.id
        }

    override var deleted: Boolean? = null
        set(deleted) {
            field = this.deleted
        }

    override val isNew: Boolean
        get() = Optional.ofNullable<I>(this.id).isPresent
}
